src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -27,11 +27,14 @@ public function __invoke(string $composerJsonPath) : Generator
 
     private function getFilePaths(array $sourceDirs, string $packageDir) : array
     {
+        $flattened = array_reduce($sourceDirs, function(array $sourceDirs, $sourceDir) {
+            return $sourceDirs + (is_array($sourceDir) ? $sourceDir : [$sourceDir]);
```

Comment:
You can cast to an array? Well, that's the way to go then. I don't think it can be anything but a [string or an array](https://getcomposer.org/doc/04-schema.md#psr-4).